roachtest: clock/jump/large_backward_enabled failed [Span use after finish]

roachtest.clock/jump/large_backward_enabled [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4389357&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4389357&tab=artifacts#/clock/jump/large_backward_enabled) on master @ [6d9ceec64fa120113c484d17e3f61dfd6737b221](https://github.com/cockroachdb/cockroach/commits/6d9ceec64fa120113c484d17e3f61dfd6737b221):


```
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:1038
		  | main.(*testImpl).Fatalf
		  | 	main/pkg/cmd/roachtest/test_impl.go:254
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.runClockJump
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/clock_jump_crash.go:86
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerClockJumpTests.func1
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/clock_jump_crash.go:143
		  | main.(*testRunner).runTest.func2
		  | 	main/pkg/cmd/roachtest/test_runner.go:779
		  | runtime.goexit
		  | 	GOROOT/src/runtime/asm_amd64.s:1581
		Wraps: (2) cluster.RunE
		Wraps: (3) context canceled
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errors.errorString

	cluster.go:1981,clock_util.go:112,clock_jump_crash.go:67,panic.go:1038,test_impl.go:248,cluster.go:1981,clock_util.go:112,panic.go:1038,test_impl.go:254,clock_jump_crash.go:86,clock_jump_crash.go:143,test_runner.go:779: cluster.RunE: context canceled
		(1) attached stack trace
		  -- stack trace:
		  | main.(*clusterImpl).RunE
		  | 	main/pkg/cmd/roachtest/cluster.go:1998
		  | main.(*clusterImpl).Run
		  | 	main/pkg/cmd/roachtest/cluster.go:1979
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.(*offsetInjector).recover
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/clock_util.go:112
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.runClockJump.func1
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/clock_jump_crash.go:67
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:1038
		  | main.(*testImpl).Fatal
		  | 	main/pkg/cmd/roachtest/test_impl.go:248
		  | main.(*clusterImpl).Run
		  | 	main/pkg/cmd/roachtest/cluster.go:1981
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.(*offsetInjector).recover
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/clock_util.go:112
		  | runtime.gopanic
		  | 	GOROOT/src/runtime/panic.go:1038
		  | main.(*testImpl).Fatalf
		  | 	main/pkg/cmd/roachtest/test_impl.go:254
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.runClockJump
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/clock_jump_crash.go:86
		  | github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests.registerClockJumpTests.func1
		  | 	github.com/cockroachdb/cockroach/pkg/cmd/roachtest/tests/clock_jump_crash.go:143
		  | main.(*testRunner).runTest.func2
		  | 	main/pkg/cmd/roachtest/test_runner.go:779
		  | runtime.goexit
		  | 	GOROOT/src/runtime/asm_amd64.s:1581
		Wraps: (2) cluster.RunE
		Wraps: (3) context canceled
		Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errors.errorString
```
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*clock/jump/large_backward_enabled.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13204

```
panic: use of Span after Finish. Span: serial unordered sync. Finish previously called at: <stack not captured. Set debugUseAfterFinish>

goroutine 124 [running]:
panic({0x4516b80, 0xc0006a51c0})
	GOROOT/src/runtime/panic.go:1147 +0x3a8 fp=0xc00ab1bb00 sp=0xc00ab1ba40 pc=0x48a768
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).detectUseAfterFinish(0x98c0c20)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:185 +0x134 fp=0xc00ab1bb68 sp=0xc00ab1bb00 pc=0xb926d4
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).RecordingType(0xc000d48000)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:387 +0x1e fp=0xc00ab1bb80 sp=0xc00ab1bb68 pc=0xb92f7e
github.com/cockroachdb/cockroach/pkg/util/tracing.(*Span).IsVerbose(...)
	github.com/cockroachdb/cockroach/pkg/util/tracing/span.go:395
github.com/cockroachdb/cockroach/pkg/util/log.getSpanOrEventLog({0x61dc2d8, 0xc0014ec1e0})
	github.com/cockroachdb/cockroach/pkg/util/log/trace.go:92 +0x3a fp=0xc00ab1bbb0 sp=0xc00ab1bb80 pc=0xcbc25a
github.com/cockroachdb/cockroach/pkg/util/log.logfDepth({0x61dc2d8, 0xc0014ec1e0}, 0x1, 0x2, 0x0, {0x4e27f16, 0x28}, {0xc00b74ce90, 0x1, 0x1})
	github.com/cockroachdb/cockroach/pkg/util/log/channels.go:55 +0x2a8 fp=0xc00ab1be20 sp=0xc00ab1bbb0 pc=0xca08a8
github.com/cockroachdb/cockroach/pkg/util/log.Warningf(...)
	github.com/cockroachdb/cockroach/bazel-out/k8-opt/bin/pkg/util/log/log_channels_generated.go:449
github.com/cockroachdb/cockroach/pkg/util/hlc.(*Clock).checkPhysicalClock(0xc0020c1718, {0x61dc2d8, 0xc0014ec1e0}, 0x1, 0x16dbe53749c2936c)
	github.com/cockroachdb/cockroach/pkg/util/hlc/hlc.go:334 +0x127 fp=0xc00ab1bee0 sp=0xc00ab1be20 pc=0xccc227
github.com/cockroachdb/cockroach/pkg/util/hlc.(*Clock).getPhysicalClockAndCheck(0xc0009a9220, {0x61dc2d8, 0xc0014ec1e0})
	github.com/cockroachdb/cockroach/pkg/util/hlc/hlc.go:319 +0x88 fp=0xc00ab1bf28 sp=0xc00ab1bee0 pc=0xccc0c8
github.com/cockroachdb/cockroach/pkg/util/hlc.(*Clock).StartMonitoringForwardClockJumps.func1()
	github.com/cockroachdb/cockroach/pkg/util/hlc/hlc.go:278 +0x1cf fp=0xc00ab1bfe0 sp=0xc00ab1bf28 pc=0xccbfaf
runtime.goexit()
	GOROOT/src/runtime/asm_amd64.s:1581 +0x1 fp=0xc00ab1bfe8 sp=0xc00ab1bfe0 pc=0x4bf181
created by github.com/cockroachdb/cockroach/pkg/util/hlc.(*Clock).StartMonitoringForwardClockJumps
	github.com/cockroachdb/cockroach/pkg/util/hlc/hlc.go:253 +0xff
```

--- a/pkg/util/hlc/BUILD.bazel
+++ b/pkg/util/hlc/BUILD.bazel
@@ -19,6 +19,7 @@ go_library(
         "//pkg/util/log",
         "//pkg/util/syncutil",
         "//pkg/util/timeutil",
+        "//pkg/util/tracing",
         "@com_github_cockroachdb_errors//:errors",
         "@com_github_cockroachdb_redact//:redact",
     ],

--- a/pkg/util/hlc/hlc.go
+++ b/pkg/util/hlc/hlc.go
@@ -18,6 +18,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"github.com/cockroachdb/cockroach/pkg/util/tracing"
 	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/redact"
 )
@@ -250,7 +251,9 @@ func (c *Clock) StartMonitoringForwardClockJumps(
 		return errors.New("clock jumps are already being monitored")
 	}
 
+	ctx, sp := tracing.ForkSpan(ctx, "clock monitor")
 	go func() {
+		defer sp.Finish()
 		// Create a ticker object which can be used in selects.
 		// This ticker is turned on / off based on forwardClockJumpCheckEnabledCh
 		ticker := tickerFn(time.Hour)

